rfd/0135-kube-apps-discovery.md

```diff
@@ -0,0 +1,234 @@
+---
+authors: Anton Miniailo (anton@goteleport.com)
+state: draft
+---
+
+# RFD 135 - Kubernetes App discovery
+
+## Required Approvers
+
+- Engineering: `@r0mant` && `@tigrato`
+- Product: `@klizhentas` || `@xinding33`
+- Security: `@reedloden` || `@jentfoo`
+
+## What
+
+Proposes discovery of apps that are hosted inside Kubernetes clusters and exposed as services.
+
+Related issue: [#25538](https://github.com/gravitational/teleport/issues/25538)
+
+## Why
+
+Some users dynamically deploy apps into their Kubernetes clusters and expose those as Kubernetes services. When they actively
+create/update/remove apps, manually trying to mirror these changes to Teleport may be time-consuming and prone to
+errors, especially for organizations that have a large number of services running in their Kubernetes clusters.
+Teleport will be able to discover Kubernetes service and automatically register them as
+Teleport apps and keep them in up to date state, simplifying process and making user experience much better.
+
```

Comment:
Also, this allows a security team to install and manage the agent while the development team deploys applications. Currently, the dev team has to sync with the security team to declare new applications each time.

([reported by a customer](https://goteleport.slack.com/archives/C0582MBSMHN/p1689937545904449): with a detailed explanation of their use case and why this is painful for them)